assets/js/index.js

```diff
@@ -198,13 +206,24 @@ function updateDisplay() {
         if (i == code) {
           selected = true;
         }
-        text += `<option value="${i}" ${i == code ? 'selected': ''}>` + messageArray[j].name + '</option>';
+        var catText = categoryMap.get(messageArray[j].category);
+        var option = `<option value="${i}"${i == code ? ' selected': ''}>` + messageArray[j].name + '</option>';
+        // Append option to category
+        catText = !catText ? option : catText + option;
+        categoryMap.set(messageArray[j].category, catText);
         // Map message to dropdown ID for later recognition
         dropdownMap.set(i, messageArray[j]);
       }
     }
   }
 
+  // Format dropdown menu labels
+  for (const category of variables.categories) {
+    if (categoryMap.get(category.category)) {
+      text += `<optgroup label="${category.value}">${categoryMap.get(category.category)}</optgroup>`
+    }
+  }
```

Comment:
Just some nitpick to use destruction so we don't have `category.category` ;p

```suggestion
  for (const { category, value } of variables.categories) {
    if (categoryMap.get(category)) {
      text += `<optgroup label="${value}">${categoryMap.get(category)}</optgroup>`
    }
  }
```